posts/how-to-make-a-composer-package.md

```diff
@@ -0,0 +1,552 @@
+
+For our next prerequisites, we will need [Docker](https://docs.docker.com/engine/install/) and [Docker Compose](https://docs.docker.com/compose/install/). These are tools that help us to run our container because we will be using them to run PHP and composer without actually installing them on our machine.
+
+You can check whether these are installed by confirming whether there is output when you check for version number. First, check that Docker is installed.
+
+<Highlighter
+input='docker -v'
+language='bash'
+/>
+
+If you receive output with a version number you are all set. Next, check if Docker Compose is installed.
+
+<Highlighter
+input='docker-compose -v'
+language='bash'
+/>
+
+Again, if you received output, then Docker Compose is installed.
+
+If you received an error message or no version number, then either or both are not installed. Head to the [Docker installation page](https://docs.docker.com/engine/install/) and/or the [Docker Compose installation page](https://docs.docker.com/compose/install/). 
+
+Finally, you will need to install [Composer](https://getcomposer.org/), a dependency manager for PHP. You can read about [installing Composer on the official download guide](https://getcomposer.org/download/). If you are on Ubuntu or Debian Linux, you can use the following commansd to install Compose from your operating system's package manager.
+
+<Highlighter
+input={`sudo apt update
+sudo apt install composer`}
+language='bash'
+/>
+
+With these prerequisites in place on your machine, you're ready to move forward with this tutorial.
+
+## Configuring the environment
+
+We'll start with configuring our environment. First, we will create a directory for our project. So, let's create a directory called `said-quotes`. After that, we need a file called `sg` inside the created folder.
+
+<Highlighter
+input={`mkdir said-quotes
+cd said-quotes/
+touch sg`}
+language='bash'
+/>
+
+This `sg` file will contain the code that we will use to run our container. So, let's add the following code to our `sg` file:
+
+<Highlighter
+input={`#!/usr/bin/env bash
+ 
+docker run -it --rm \
+    --user $(id -u):$(id -g) \
+    -v $PWD:/app \
+    -w /app composer:latest "$@"`}
+language='bash'
+/>
+
+This code runs a Composer container inside our project folder. The `--user` flag tells the container to run as the current user. The `-v` flag tells the container to mount the current directory to the container. The `-w` flag tells the container to run it in the current directory.
+
+Now, we need to give the `sg` file the right permissions so that it can be executed.
+
+<Highlighter
+input='chmod +x sg'
+language='bash'
+/>
+
+With the environment configured, we can move onto initializing our project.
+
+## Initializing the project
+
+We can now start creating our project. Let's run the following command:
+
+<Highlighter
+input='sg init'
```

Comment:
I am unable to get this command to run on a clean Linux server. I added Composer to the prerequisites but this is not running as an executable, is there something else that needed to be set up first?

Reply:
Oh, sorry about that! I think I forgot to put the symbol to point to the current directory, so, the command is actually:

```bash
./sg init
```

just `sg` will run nothing cause is not a real command or alias in the bash.

![image](https://user-images.githubusercontent.com/18440704/142962879-2b01d395-18ca-47ec-9828-13866286d247.png)